ledger/src/test_helpers/chain_builder.rs

```diff
@@ -181,32 +222,69 @@ impl<N: Network> TestChainBuilder<N> {
             last_block_round: 0,
             round_to_certificates: Default::default(),
             previous_leader_certificate: Default::default(),
+            transactions,
         })
     }
 
     /// Create multiple blocks, with fully-connected DAGs.
     pub fn generate_blocks(&mut self, num_blocks: usize, rng: &mut TestRng) -> Result<Vec<Block<N>>> {
-        self.generate_blocks_with_opts(num_blocks, GenerateBlocksOptions::default(), rng)
+        let num_validators = self.private_keys.len();
+        let transactions = mem::take(&mut self.transactions);
+
+        self.generate_blocks_with_opts(
+            num_blocks,
+            GenerateBlocksOptions { num_validators, transactions, ..Default::default() },
```

Comment:
What if not all transactions fit in the block(s)? Will the rest be stored back in `self.transactions`?